src/OrchardCore/OrchardCore.Workflows.Abstractions/Activities/Activity.cs

```diff
@@ -146,6 +146,6 @@ protected virtual T GetProperty<T>(Type type, Func<T> defaultValue = null, [Call
     protected virtual void SetProperty(object value, [CallerMemberName] string name = null)
     {
         // Properties[name] = JToken.FromObject(value);
-        Properties[name] = JNode.FromObject(value);
+        Properties[name] = JNode.FromObject(value).DeepClone();;
```

Comment:
```suggestion
        Properties[name] = value is JsonNode node ? node.DeepClone() : JNode.FromObject(value);
```
Isn't it enough to clone the node if it already is one? Otherwise it's a freshly serialized object with no parent, isn't it?